### PR TITLE
Setting up auth app to get user in request while creating views

### PR DESCRIPTION
**Pull request by Shwetabhk**

### CODE REVIEW

File: apps/fyle_rest_auth/views.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+
+class LoginView(APIView):
+    """
+    Login Using Fyle Account
+    """
+    authentication_classes = []
+    permission_classes = []
+
+    def post(self, request):
```
Comment (Shwetabhk) [inline apps/fyle_rest_auth/views.py]: This takes authorization `code` returns `refresh_token` and `access_token`.
The url is `/login`, creates a new user according to email if it doesn't already exist
Postman

File: apps/fyle_rest_auth/views.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        except InternalServerError:
+            return Response(
+                {
+                    'message': 'Wrong/Expired Authorization code'
+                },
+                status=status.HTTP_401_UNAUTHORIZED
+            )
+
+
+class RefreshView(APIView):
```
Comment (Shwetabhk) [inline apps/fyle_rest_auth/views.py]: This takes `refresh_token` and returns a new access token
The URL is `/refresh`
Postman

File: apps/fyle_rest_auth/models.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from django.db import models
+from django.contrib.auth import get_user_model
+
+User = get_user_model()
+
+
+class AuthTokens(models.Model):
```
Comment (Shwetabhk) [inline apps/fyle_rest_auth/models.py]: Creating a separate table to save tokens, to keep it independent from the User Model

File: apps/fyle_rest_auth/urls.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
+Including another URLconf
+    1. Import the include() function: from django.urls import include, path
+    2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
+"""
+from django.urls import path
+
+from .views import LoginView, RefreshView
+
+urlpatterns = [
```
Comment (Shwetabhk) [inline apps/fyle_rest_auth/urls.py]: can change urls if required

File: apps/fyle_rest_auth/utils.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
+"""
+Authentication utils
```
Comment (Shwetabhk) [inline apps/fyle_rest_auth/utils.py]: Auth utility functions

**Comment by Shwetabhk** [comment]
This authentication app at this very moment is separatable into another app, which can be released on PyPi

File: apps/fyle_rest_auth/models.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+from django.db import models
+from django.contrib.auth import get_user_model
+
+User = get_user_model()
+
+
+class AuthToken(models.Model):
```
Comment (Shwetabhk) [inline apps/fyle_rest_auth/models.py]: Creating a separate table to save tokens, to keep it independent from the User Model

File: apps/fyle_rest_auth/views.py [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+            if not user:
+                return Response(
+                    {
+                        'message': 'User record not found, please login'
+                    },
+                    status=status.HTTP_401_UNAUTHORIZED
+                )
+
+            auth_token = AuthToken.objects.get(user=user)
+            auth_token.access_token = tokens['access_token']
```
Comment (snarayanank2) [inline apps/fyle_rest_auth/views.py]: why is access_token stored and updated in the model?

Comment (Shwetabhk) [reply apps/fyle_rest_auth/views.py]: That is actually not needed, will change it

**Pull request merged**